File: src/main/java/io/github/smootheez/curseforge/CurseforgeMetadata.java
package io.github.smootheez.curseforge;

import lombok.*;

import java.util.List;

/**
 * Represents the metadata payload sent when uploading a file to CurseForge.
 *
 * @param changelog                  the changelog text for this upload
 * @param changelogType              the format used to interpret the changelog (text, markdown, html)
 * @param displayName                the display name of the uploaded file
 * @param gameVersions               numeric CurseForge game version identifiers
 * @param releaseType                the release channel (release, beta, alpha)
 * @param isMarkedForManualRelease   whether the upload requires manual approval
 * @param relations                  dependency relationships declared for this upload
 */
@Builder
public record CurseforgeMetadata(

        String changelog,

        ChangelogType changelogType,

        String displayName,

        List<Integer> gameVersions,

        String releaseType,

        boolean isMarkedForManualRelease,

        Projects relations
) { }

File: src/main/kotlin/io/github/smootheez/curseforge/CurseforgeMetadata.kt
package io.github.smootheez.curseforge

import kotlinx.serialization.Serializable

@Serializable
data class CurseforgeMetadata(
//    val projectId: String, not include the project id because project id is requested by parameter
    val changelog: String,
    val changelogType: ChangelogType,
    val displayName: String? = null,
//    val parentFileID: String? = null, don't need parentID
    val gameVersions: List<Int>,
    val releaseType: String, //
    val isMarkedForManualRelease: Boolean,
    val relations: Projects
)

@Serializable
data class Projects(
    val projects: List<ProjectsMetadata> = emptyList()
)

@Serializable
data class ProjectsMetadata(
    val projectID: Int?,
    val relationType: RelationType
)

